substrate/frame/support/src/lib.rs

```diff
@@ -1111,23 +1111,37 @@ pub mod pallet_macros {
 	/// #
 	/// 	#[pallet::pallet]
 	/// 	pub struct Pallet<T>(_);
+	///
+	///  	#[pallet::config]
+	///  	pub trait Config: frame_system::Config {
+    ///         /// Type for specifying dispatchable weights.
+    ///         type WeightInfo: WeightInfo;
+    ///     }
 	///
 	/// 	#[pallet::call]
 	/// 	impl<T: Config> Pallet<T> {
-	/// 		#[pallet::weight({0})] // <- set actual weight here
+	/// 		#[pallet::weight(<T as Config>::WeightInfo::do_something())] 
 	/// 		#[pallet::call_index(0)]
-	/// 		pub fn something(
-	/// 			_: OriginFor<T>,
+	/// 		pub fn do_something(
+	/// 			origin: OriginFor<T>,
 	/// 			foo: u32,
 	/// 		) -> DispatchResult {
-	/// 			unimplemented!()
+	/// 			Ok(())
 	/// 		}
 	/// 	}
-	/// #
-	/// # 	#[pallet::config]
-	/// # 	pub trait Config: frame_system::Config {}
 	/// }
-	/// ```
+	///
+	/// /// The `WeightInfo` trait defines weight functions for dispatchable calls.
+    /// pub trait WeightInfo {
+    ///     fn do_something() -> Weight;
+    /// }
+    /// ```
+    /// In this example:
+    ///
+    /// - The `WeightInfo` trait defines a function `do_something` that returns the weight for the corresponding dispatchable.
+    /// - The `Config` trait includes an associated type `WeightInfo` that implements the `WeightInfo` trait.
+    /// - The `#[pallet::weight(<T as Config>::WeightInfo::do_something())]` attribute assigns the weight to the `do_something` dispatchable function.
```

Comment:
we need to mention that the weight can be implicitly defined.
When not explicitly defined, the macro will lookup to get the weightinfo defined in the `pallet::call` attribute, and use weight info :: function name.